web/src/message_view_header.js

```diff
@@ -2,13 +2,18 @@ import $ from "jquery";
 
 import render_message_view_header from "../templates/message_view_header.hbs";
 
+import {Filter} from "./filter";
 import {$t} from "./i18n";
 import * as inbox_util from "./inbox_util";
 import * as narrow_state from "./narrow_state";
 import * as peer_data from "./peer_data";
 import * as popovers from "./popovers";
 import * as recent_view_util from "./recent_view_util";
 import * as rendered_markdown from "./rendered_markdown";
+import * as search from "./search";
+import * as search_pill from "./search_pill";
+
+// put widget in its own file to avoid circular imports?
```

Comment:
Or maybe... we move all the search related code in `message_view_header.js` to `search.js`, such that `message_view_header` can import `search` instead of vice versa.

https://chat.zulip.org/#narrow/stream/6-frontend/topic/search.20code.20files